Dev

Webkonditer opened the pull request.

AIvachkin committed:
Создана кнопка меню (без отклика бота), класс инициализации. Добавлены комментарии

AIvachkin committed:
Создана кнопка меню (без отклика бота), класс инициализации. Добавлены комментарии

DenisTr2005 committed:
Обработана команда /start - выводится приветственное сообщение.

DenisTr2005 committed:
Обработана команда /start - выводится приветственное сообщение
Комментарии с /**
StartCommand вынесен в поле и конструктор с аннотацией @Lazy

AIvachkin committed:
Merge pull request #3 from AIvachkin/feature_Denis

Обработана команда /start - выводится приветственное сообщение

AIvachkin committed:
Merge branch 'feature_Andrew' into dev

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java

AIvachkin committed:
Сощзданы классы: User, UserServicу и UserRepository. Не работает инициализация поля "phoneNumber" (не видит сеттер), не работает сохранение в БД с помощью метода .save

DenisTr2005 committed:
класс Pet с полями id, name, age, trustee(one to one)
репозиторий к нему
класс PetService и метод в нем createPet, который сохраняет животное в БД

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Alexander

# Conflicts:
#	pom.xml
#	src/main/java/pro/sky/JD2AnimalShelterBot/model/Pet.java

Webkonditer committed:
dev1

Webkonditer committed:
Собрал из всех наших кусочков рабочую версию

AIvachkin committed:
Merge pull request #7 from AIvachkin/feature_Alexander

Feature alexander

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/configuration/BotInitializer.java:
```diff
@@ -0,0 +1,42 @@
+     * Поле типа TelegramBot для подключения нашего приложения к telegram
+     */
+    @Autowired
+    TelegramBot bot;
```
private

AIvachkin replied:
исправили

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/configuration/BotInitializer.java:
```diff
@@ -0,0 +1,42 @@
+    /**
+     * Поле типа TelegramBot для подключения нашего приложения к telegram
+     */
+    @Autowired
```
инъекции лучше через конструкторы

AIvachkin replied:
исправили

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/configuration/BotInitializer.java:
```diff
@@ -0,0 +1,42 @@
+     *
+     * @throws TelegramApiException - исключение в случае прихода неизвестной для бота команды
+     */
+    @EventListener({ContextRefreshedEvent.class})
```
а зачем здесь event listener?

AIvachkin replied:
чтобы получать уведомления о новом событии/запросе

dmitry-bizin replied:
ну это спринговый эвент и его спринг выкидывает при рефреше контекста, а как рефреш контекста связан с получением нового апдейта от пользователя?

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,123 @@
+     *                      дополнительно создается меню для бота
+     *                      listOfCommands - лист, содержащий команды меню
+     */
+    public TelegramBot(@Lazy StartCommand startCommand, BotConfiguration configuration) {
```
lazy инициализация говорит о том, что есть циклическая зависимость бинов, что обычно следствие бедной архитектуры кода. Это область для рефакторинга, подумайте над этим

AIvachkin replied:
Взяли в работу

dmitry-bizin commented on src/main/resources/application.properties:
```diff
@@ -1,2 +1,11 @@
 bot.token=5867119344:AAHNECKmgfXDZ4Ytw12kBPd6w6kuC443Dyw
+
+#db related settings
+spring.jpa.hibernate.ddl-auto=update
```
таблицы лучше создавать через ликвибейз миграции, а здесь использовать стратегию validate, чтобы hibernate не создавал/обновлял, а только валидировал

AIvachkin replied:
исправлено

Webkonditer committed:
Добавил сущности для переписки и отчетов попечителей. Сделал Пет контроллер. Подключил Сваггер.

Webkonditer committed:
Добавлены комментарии в Пет сервисе к новым методам.

Webkonditer committed:
Добавлены репозитории к новым сущностям.

Webkonditer committed:
Добавлены репозитории к новым сущностям.

Webkonditer committed:
Сделаны два метода в волонтерском контроллере.

DenisTr2005 committed:
Подключил liquibase, создал таблицы

Webkonditer committed:
Сделаны два эксершена в пет контроллере.

Webkonditer committed:
Сделаны два метода в волонтерском контроллере:
Метод для отправки сообщения пользователю
Метод для получения всей переписки с пользователем

AIvachkin committed:
Создан класс TakePet, метод onButton(),список команд для него, а также методы, обрабатывающие данные команды

Webkonditer committed:
Исправлен пост на гет в волонтерском контроллере

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Alexander

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Denis' into feature_Alexander

AIvachkin committed:
Merge pull request #11 from AIvachkin/feature_Denis

Подключен ликвибейз и созданы таблицы: pet, user, volunteer, report

AIvachkin committed:
Merge pull request #10 from AIvachkin/feature_Andrew

Создан класс TakePet, метод onButton(),список команд для него, а такж…

AIvachkin committed:
Merge pull request #9 from AIvachkin/feature_Alexander

Feature alexander

Webkonditer committed:
Ссделал кнопочное меню первого уровня. Добавил нужные таблицы в ликвибейз.

AIvachkin committed:
К контроллерам написана документация через аннотации OpenAPI

DenisTr2005 committed:
добавил таблицы correspondence и trustees_reports

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/model/TrusteesReports.java:
```diff
@@ -0,0 +1,62 @@
+    /**
+     * ИД животного
+     */
+    private Long petId;
```
здесь должен быть объект Pet, чтобы связь была

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/model/TrusteesReports.java:
```diff
@@ -0,0 +1,62 @@
+     */
+    @Id
+    @GeneratedValue
+    private Long messageId;
```
здесь лучше просто id

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/model/Correspondence.java:
```diff
@@ -0,0 +1,55 @@
+     */
+    @Id
+    @GeneratedValue
+    private Long messageId;
```
здесь лучше просто id

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/model/Pet.java:
```diff
@@ -0,0 +1,56 @@
+     */
+    @Id
+    @GeneratedValue(strategy=GenerationType.IDENTITY)
+    private Long petId;
```
здесь лучше просто id

dmitry-bizin commented on src/main/resources/liquibase/scripts/changesets.sql:
```diff
@@ -0,0 +1,48 @@
+);
+-- changeset denis:3
+CREATE TABLE trustees_reports (
+    message_id BIGINT,
```
id

dmitry-bizin commented on src/main/resources/liquibase/scripts/changesets.sql:
```diff
@@ -0,0 +1,48 @@
+
+-- changeset denis:2
+CREATE TABLE correspondence (
+    message_id BIGINT,
```
id

dmitry-bizin commented on src/main/resources/liquibase/scripts/changesets.sql:
```diff
@@ -0,0 +1,48 @@
+
+-- changeset denis:1
+CREATE TABLE pet (
+    pet_id BIGINT,
```
id

dmitry-bizin commented on src/main/resources/liquibase/scripts/changesets.sql:
```diff
@@ -0,0 +1,48 @@
+    phone_number VARCHAR
+);
+CREATE TABLE volunteer (
+    vol_id BIGINT,
```
id

dmitry-bizin commented on src/main/resources/liquibase/scripts/changesets.sql:
```diff
@@ -0,0 +1,48 @@
+    lastname VARCHAR,
+    phone_number VARCHAR
+);
+CREATE TABLE report (
```
для этой таблицы пока что нет класса-сущности

Webkonditer replied:
таблица дропнута ниже.

dmitry-bizin replied:
а зачем тогда создавали?)

Webkonditer replied:
Не договорились по-началу.

dmitry-bizin commented on src/main/resources/liquibase/scripts/changesets.sql:
```diff
@@ -0,0 +1,48 @@
+    lastname VARCHAR,
+    phone_number VARCHAR
+);
+CREATE TABLE volunteer (
```
для этой таблицы пока что нет класса-сущности

Webkonditer replied:
Решили не делать таблицу волонтеров. Вся волонтерская работа будет через API. Таблица дропнута ниже.

dmitry-bizin replied:
тогда создание таблицы нужно бы убрать

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,113 @@
+     * @param pet
+     * @return
+     */
+    @PostMapping
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,113 @@
+     * @param petId
+     * @return
+     */
+    @GetMapping("{petId}")
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,113 @@
+     * @param pet
+     * @return
+     */
+    @PutMapping()
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,113 @@
+     * @param petId
+     * @return
+     */
+    @DeleteMapping("/{petId}")
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,113 @@
+     * Метод для получения списка всех животных
+     * @return
+     */
+    @GetMapping("/all")
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,113 @@
+     * @param chatId идентификатор попечителя
+     * @return
+     */
+    @GetMapping("/assign")
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/PetController.java:
```diff
@@ -0,0 +1,113 @@
+     * @param petId идетификатор животного
+     * @return
+     */
+    @GetMapping("/detach")
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/VolunteerController.java:
```diff
@@ -0,0 +1,85 @@
+     *Метод для просмотра всех неотвеченных сообщений
+     * @return
+     */
+    @GetMapping("/messages")
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/VolunteerController.java:
```diff
@@ -0,0 +1,85 @@
+     *Метод для отправки ответа на сообщение
+     * @return
+     */
+    @PostMapping("/messages/{messageId}/answer")
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/controller/VolunteerController.java:
```diff
@@ -0,0 +1,85 @@
+     *Метод для получения всей переписки с пользователем
+     * @return
+     */
+    @PostMapping("/messages/{chatId}")
```
нужно использовать ещё аннотации OpenApi для описания эндпоинтов

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,123 @@
+    /**
+     * Поле - конфигурация: для работы методов по получению имени бота и его токена
+     */
+    final BotConfiguration configuration;
```
private

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Alexander

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/controller/VolunteerController.java

AIvachkin committed:
Merge branch 'feature_Alexander' into feature_Andrew

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/controller/VolunteerController.java

Webkonditer committed:
Merge remote-tracking branch 'origin/dev' into feature_Alexander

# Conflicts:
#	src/main/resources/liquibase/scripts/changesets.sql

AIvachkin committed:
К контроллерам написана документация через аннотации OpenAPI + добавлены описания к методам PETS (Put и Post)

Webkonditer committed:
Ссделал первый тест в CorrespondenceService. Подрихтовал таблицы в liquebase.

AIvachkin committed:
Merge branch 'feature_Alexander' into feature_Andrew

Webkonditer committed:
Ссделал второй тест в CorrespondenceService.

Webkonditer committed:
Ссделал третий тест в CorrespondenceService.

AIvachkin committed:
Merge branch 'feature_Alexander' into feature_Andrew

Webkonditer committed:
Ссделал все тесты в CorrespondenceService.

Webkonditer committed:
Заменил @EventListener({ContextRefreshedEvent.class}) на @PostConstruct в BotInitializer

AIvachkin committed:
Добавлен тест для класса UserService (не работает)

DenisTr2005 committed:
Класс ShelterInfo + обработка нажатия на кнопки меню - рабочая версия бота

Webkonditer committed:
Merge pull request #12 from AIvachkin/feature_Alexander

Feature alexander

AIvachkin committed:
Добавлен тест для класса UserService - работает

AIvachkin committed:
Написаны тесты для UserService

AIvachkin committed:
Написаны тесты для TakePet

AIvachkin committed:
В классе TakePet написан метод для формирования клавиатуры (кнопок)

Webkonditer committed:
Создан класс контекста пользователя.
Бот начал отвечать на кнопку "Позвать волонтера".

AIvachkin committed:
В классе TakePet дописал команды для получения информационной справки. Исправил ошибки с передачей параметра в метод, формирующий ответ на запрос

Webkonditer committed:
Закрыл тестами класс Commands.

Webkonditer committed:
Закрыл тестами класс Commands.

Webkonditer committed:
Сделал первый тест в VolunteerController.

DenisTr2005 committed:
Создан класс ExecuteMessage для отправки сообщений
Класс ShelterInfo доработан с созданием кнопочного меню
Рабочая версия бота

DenisTr2005 committed:
Merge remote-tracking branch 'origin/feature_Denis' into feature_Denis

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/ShelterInfo.java
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/TakePet.java
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java

DenisTr2005 committed:
Создан класс ExecuteMessage для отправки сообщений
Класс ShelterInfo доработан с созданием кнопочного меню
Рабочая версия бота

DenisTr2005 committed:
Создан класс ExecuteMessage для отправки сообщений
Удален класс OnButton
Класс ShelterInfo доработан с созданием кнопочного меню
Рабочая версия бота

AIvachkin committed:
Написаны 2 теста для класса StartCommand

Webkonditer committed:
Сделал все тесты в VolunteerController.

Webkonditer committed:
Merge pull request #16 from AIvachkin/feature_Alexander

Создан класс контекста пользователя.

AIvachkin committed:
Merge pull request #15 from AIvachkin/feature_Andrew

Feature andrew

AIvachkin committed:
Merge branch 'dev' into feature_Denis

AIvachkin committed:
Merge pull request #17 from AIvachkin/feature_Denis

feature_Denis

DenisTr2005 committed:
PetControllerTest 100%

AIvachkin committed:
Исправлены тесты для классов StartCommand и UserService

AIvachkin committed:
Исправлен тест для класса TakePet

AIvachkin committed:
Перенес команды (ENUM) из TakePet в Commands. Перегруппировал кнопки меню (TakePet). Обновил тест для класса TakePet

Webkonditer committed:
Сделана фича по отправке сообщения волонтеру с запросом и сохранением телефона пользователя.

AIvachkin committed:
Merge pull request #18 from AIvachkin/feature_Denis

PetControllerTest

dmitry-bizin commented on pom.xml:
```diff
@@ -1,55 +1,103 @@
+
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
```
Эта зависимость лишняя

dmitry-bizin commented on pom.xml:
```diff
@@ -1,55 +1,103 @@
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
```
эта тоже

dmitry-bizin commented on pom.xml:
```diff
@@ -1,55 +1,103 @@
+        </dependency>
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
```
и эта (второй раз)

BizinMitya commented on src/main/java/pro/sky/JD2AnimalShelterBot/model/Correspondence.java:
```diff
@@ -0,0 +1,54 @@
+    private boolean answered;
+
+    /**
+     * пользователь или волонтер
```
вам нужно будет иметь некую логику для волонтёров (позвать для общения), поэтому желательно создать таблицу волонтёров, а здесь внешний ключ на волонтёра

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/model/TrusteesReports.java:
```diff
@@ -0,0 +1,59 @@
+    /**
+     * фото животного
+     */
+    private String photo;
```
также не совсем понятно, в каком виде у вас фото? лучше создать отдельную таблицу для хранения фото, а здесь будет само фото уже по внешнему ключу

Webkonditer replied:
Здесь будет ссылка на фото на нашем серваке.

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,153 @@
+     *                      дополнительно создается меню для бота
+     *                      listOfCommands - лист, содержащий команды меню
+     */
+    public TelegramBot(@Lazy StartCommand startCommand,
```
Lazy аннотации признак сильной связности бинов, что не есть хорошо.. Постарайтесь уйти от двунаправленной связи бинов. Вам нужно выделить независимую функциональность в бин, который будут остальные сервисы использовать

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,153 @@
+                    startCommand.startCallBack(chatId, update);
+                    break;
+
+                case "❓ Узнать информацию о приюте":
```
здесь лучше использовать константы

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,153 @@
+            //Commands.CALL_VOLUNTEER_COMAND.getLabel();
+// оператор выбора будет дописан позже после получения полного набора команд
+            switch (messageText) {
+                case "/start":
```
здесь лучше использовать константы

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,153 @@
+                case "❓ Узнать информацию о приюте":
+                    shelterInfo.createMenuShelterInfo(chatId);
+                    break;
+                case "\uD83D\uDC36️ Как взять собаку из приюта":
```
здесь лучше использовать константы

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java:
```diff
@@ -0,0 +1,153 @@
+                case "\uD83D\uDC36️ Как взять собаку из приюта":
+                    takePet.takePetCommandReceived(chatId);
+
+                case "\uD83D\uDC69\u200D\uD83C\uDF3E  Позвать волонтера":
```
здесь лучше использовать константы

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/util/StringUtil.java:
```diff
@@ -0,0 +1,39 @@
+
+    public static String serialize(Object o) {
+        try {
+            return new ObjectMapper().writeValueAsString(o);
```
ObjectMapper лучше сделать статик-полем, чтобы не создавать каждый раз

Webkonditer replied:
пока в технический долг

Webkonditer committed:
Сделана фича по отправке сообщения волонтеру с запросом и сохранением телефона пользователя.

AIvachkin committed:
Создан хендлер, перенесен блок обработки из Класса TelegramBot в хендлер, прописан switch для класса TakePet, скорректирована константа (рекомендации по обустройству дома для взрослой собаки)

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_andrew' into feature_Alexander

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/TelegramBot.java
#	src/test/java/pro/sky/JD2AnimalShelterBot/service/StartCommandTest.java
#	src/test/java/pro/sky/JD2AnimalShelterBot/service/TakePetTest.java

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Denis' into feature_Alexander

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/TakePet.java
#	src/test/java/pro/sky/JD2AnimalShelterBot/service/StartCommandTest.java
#	src/test/java/pro/sky/JD2AnimalShelterBot/service/TakePetTest.java

Webkonditer committed:
Подружил все ветки

AIvachkin committed:
Merge pull request #19 from AIvachkin/feature_Alexander

Feature alexander

Webkonditer committed:
Иизменил юзер контекст. Поставил сет в значение.

Webkonditer committed:
Сделал первое меню с выбором собак или кошек с записью выбора в юзер контекст.

AIvachkin committed:
Созданы классы для констант, написаны константы для класса Cat, добавлены константы для ENUM, вынесены коды для формирования кнопок в классах в отдельные методы, Создан интерфейс TakePet (его реализуют классы TakeDog и TakeCat), в хендлер добавлены константы для свича, переименованы некоторые классы и константы в связи с введением кошек

Webkonditer committed:
Сделал правки по Диминым замечаниям. Кое-что оставил с аргументацией.

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Alexander

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/handlers/UpdateHandler.java
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/StartCommand.java

Webkonditer committed:
Временный коммит

Webkonditer committed:
Временный коммит

Webkonditer committed:
Временный коммит

Webkonditer committed:
Рабочая версия, смержены все ветки на 14.12

Webkonditer committed:
Тесты PetService
Отправка фото яндекс карты местарасположения приюта

Webkonditer committed:
Добавил таблицы для собак и кошек, соответствующие модели и методы.
Все работает.

AIvachkin committed:
Merge pull request #22 from AIvachkin/feature_Alexander

Feature alexander

AIvachkin committed:
Дописаны константы для Cat, проведен рефакторинг в хендлере, реализован ответ пользователю при работе в меню для приюта для кошек, добавлена информация для приюта для кошек

DenisTr2005 committed:
Рабочая версия, смержены все ветки на 14.12
Оставить контакт в Инфо о приюте

DenisTr2005 committed:
Тесты PetService
Отправка фото яндекс карты местарасположения приюта

DenisTr2005 committed:
Тесты PetService
Отправка фото яндекс карты местарасположения приюта

AIvachkin committed:
Написан метод для сохранения контактных данных пользователя в БД при вызове данного пункта меню в приюте для кошек. Но не сохраняет номер телефона - выдает ошибку. Надо править. Способ пока не нашел.

DenisTr2005 committed:
Тесты PetService
Отправка фото яндекс карты местарасположения приюта

AIvachkin committed:
Merge branch 'feature_Alexander' into feature_Andrew

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/handlers/UpdateHandler.java
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/ShelterInfo.java
#	src/main/java/pro/sky/JD2AnimalShelterBot/сonstants/ShelterConstants.java

AIvachkin committed:
Merge pull request #24 from AIvachkin/feature_Alexander

feature_Alexander

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Alexander

AIvachkin committed:
Исправил тесты для класса StartCommand, написаны тесты для классов TakeCat и ShelterInfo

Webkonditer committed:
Сделал отдельную регистрацию для кет и дог юзеров.
Сделал переключение между приютами в боковом меню.
Исправил методы отсылки сообщений волонтеру под разные приюты.

AIvachkin committed:
- Бот может выдать контактные данные охраны для оформления пропуска на машину.
- Бот может выдать общие рекомендации о технике безопасности на территории приюта.

Webkonditer committed:
Времпнный коммиит

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Alexander

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/handlers/UpdateHandler.java
#	src/main/java/pro/sky/JD2AnimalShelterBot/сonstants/MainMenuConstants.java

Webkonditer committed:
Времпнный коммиит

Webkonditer committed:
Добавил столбец в Пет

AIvachkin committed:
Скорректирована таблица для отчетов пользователей

AIvachkin committed:
Merge branch 'feature_Alexander' into feature_Andrew

# Conflicts:
#	src/main/resources/liquibase/scripts/changesets.sql

Webkonditer committed:
Привел все API в соответсвие с кошками и собаками. Сделал несколько методов для волонтера.

dmitry-bizin commented on pom.xml:
```diff
@@ -1,55 +1,92 @@
+
+        <dependency>
+            <groupId>com.github.pengrad</groupId>
+            <artifactId>java-telegram-bot-api</artifactId>
```
зачем вам 2 разных библиотеки по работе с телеграмом?

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/model/Pet.java:
```diff
@@ -0,0 +1,63 @@
+    @JsonIgnore
+    @ManyToOne
+    @JoinColumn(name = "user_id")
+    private DogUser dogUser;
```
а почему у вас здесь DogUser, а не CatUser? Тогда нужно 2 поля - DogUser и CatUser

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/model/CatUser.java:
```diff
@@ -0,0 +1,51 @@
+    /**
+     * Переменная - питомец, которого приютил пользователь
+     */
+    @OneToMany(mappedBy = "dogUser")
```
catUser ?

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/CallVolunteer.java:
```diff
@@ -0,0 +1,18 @@
+        this.executeMessage = executeMessage;
+    }
+
+    public static void callVolunteer() {
```
static не нужно

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/UserService.java:
```diff
@@ -0,0 +1,180 @@
+     * @return
+     */
+    public List<CatUser> getAllCatUsers() {
+        return (List)catUserRepository.findAll();
```
зачем здесь каст? он не нужен, репозиторий же типизирован

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/UserService.java:
```diff
@@ -0,0 +1,180 @@
+     * @return
+     */
+    public List<DogUser> getAllDogUsers() {
+        return (List)dogUserRepository.findAll();
```
зачем здесь каст? он не нужен, репозиторий же типизирован

dmitry-bizin commented on src/main/java/pro/sky/JD2AnimalShelterBot/service/YMap.java:
```diff
@@ -0,0 +1,52 @@
+    }
+
+    /**
+     * Метод отправки карты Яндекс в чат
```
интересная задумка 👍

dmitry-bizin replied:
хотя можно без яндекс карт отправить координаты места, через SendVenue и SendLocation)

dmitry-bizin replied:
у телеграма есть такие методы

AIvachkin committed:
Добавлено поле "тип питомца" в таблицу с отчетами. Добавлены начальные команды и константы для работы с отчетами пользователей. Добавлены новые контексты (DogReport и CatReport)

AIvachkin committed:
Доделан метод по отправки карты (местоположения) для приюта кошек. Доделан метод по отправке формы ежедневного отчета пользователю (+ клавиатура для получения этой формы).

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Alexander

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/service/TrusteesReportsService.java

Webkonditer committed:
Добил все API методы для волонтера.

AIvachkin committed:
Ошибка

Webkonditer committed:
Сделвл все тесты в волонтерском контроллере

AIvachkin committed:
Начал писать методы по обработке входящего отчета пользователя

Webkonditer committed:
Убрал лишние лейзи из хандлера

Webkonditer committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Alexander

Webkonditer committed:
Служебный коммит

Webkonditer committed:
Прием и обработка фото из отчета + проверки на отсутствие фото и текста

Webkonditer committed:
Прием и обработка фото из отчета + проверки на отсутствие фото и текста

AIvachkin committed:
Дописан метод по получению и сохранению в БД отчета пользователя

AIvachkin committed:
Сервисы перегруппированы по разным папкам

AIvachkin committed:
Написаны методы для получения непрочитанных отчетов и для изменения статуса отчетов на "прочитанный"

DenisTr2005 committed:
Класс BadUser должников по отчетам
Класс CheckReports проверки своевременности отчетов

Webkonditer committed:
Сделал кучу тестов

AIvachkin committed:
Merge branch 'feature_Alexander' into feature_Andrew

AIvachkin committed:
Ветка со всеми изменениями

AIvachkin committed:
Сделаны тесты для StartCommand

DenisTr2005 committed:
Класс BadUser должников по отчетам
Класс CheckReports проверки своевременности отчетов
Таблица bad_user_data_table
BadUserRepository
BadUserService

AIvachkin committed:
Сделаны тесты для UpdateHandler

DenisTr2005 committed:
Класс BadUser должников по отчетам
Класс CheckReports проверки своевременности отчетов
Таблица bad_user_data_table
BadUserRepository
BadUserService

DenisTr2005 committed:
Класс BadUser должников по отчетам
Класс CheckReports проверки своевременности отчетов
Таблица bad_user_data_table
BadUserRepository
BadUserService

DenisTr2005 committed:
Merge remote-tracking branch 'origin/feature_Andrew' into feature_Andrew

Webkonditer committed:
Сделал свои тесты

AIvachkin committed:
Начал писать тесты для TRS

AIvachkin committed:
Merge branch 'feature_Alexander' into feature_Andrew

# Conflicts:
#	src/test/java/pro/sky/JD2AnimalShelterBot/service/Pet/PetServiceTest.java
#	src/test/java/pro/sky/JD2AnimalShelterBot/service/UserServiceTest.java

AIvachkin committed:
Tесты для TRS - 75%

AIvachkin committed:
Написан API метод по удалению пользователя из списка должников

DenisTr2005 committed:
BadUserTest

Webkonditer committed:
Сделал метод API для получения списка должников.

AIvachkin committed:
Merge branch 'feature_Denis' into feature_Andrew

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/model/BadUser.java

AIvachkin committed:
Merge branch 'feature_Alexander' into feature_Andrew

# Conflicts:
#	src/main/java/pro/sky/JD2AnimalShelterBot/controller/VolunteerController.java
#	src/main/java/pro/sky/JD2AnimalShelterBot/model/BadUser.java
#	src/test/java/pro/sky/JD2AnimalShelterBot/controller/VolunteerControllerTest.java

AIvachkin committed:
Рабочая версия 1.0

AIvachkin committed:
Рабочая версия 1.1

DenisTr2005 committed:
Добавлена аннтотация @Transactional в класс CheckReports

DenisTr2005 committed:
Исправлены замечания по CheckReports

AIvachkin committed:
Рабочая версия 1.1
Исправлены мелкие недочеты_1 часть

AIvachkin committed:
добавлен javadoc

AIvachkin committed:
добавлен javadoc_переименован пакет на docs

Webkonditer committed:
Исправил дженерики в контроллерах.

Webkonditer committed:
Исправил замечания Димы.

Webkonditer committed:
Merge pull request #30 from AIvachkin/feature_Alexander

Feature alexander
